### PR TITLE
Add docker-publish Github Action to main

### PR DESCRIPTION
**Pull request by fearnworks**
We need to add the default build action before it can be run on branches with modifications

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
+        uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 #v3.5.0
+        with:
+          cosign-release: 'v2.2.4'
```
Comment (CheesyLaZanya) [inline .github/workflows/docker-publish.yml:45]: It looks like the latest version is v2.4.1

https://github.com/marketplace/actions/cosign-installer
https://github.com/sigstore/cosign-installer?tab=readme-ov-file#usage

Any reason to be on 2.2.4?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+      # Set up BuildKit Docker container builder to be able to build
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
```
Comment (CheesyLaZanya) [inline .github/workflows/docker-publish.yml:51]: Why use the commit ID vs @v3?

https://github.com/marketplace/actions/docker-setup-buildx#usage

File: .github/workflows/docker-publish.yml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
+        uses: sigstore/cosign-installer@59acb6260d9c0ba8f4a2f9d9b48431a222b68e20 #v3.5.0
```
[the remark this answers is not in the document]
Comment (fearnworks) [reply .github/workflows/docker-publish.yml:43]: I think this was just a pin from the template. I'll try with the up to date current versions once I confirm it working with the provided checkpoints.

File: .github/workflows/docker-publish.yml [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      # multi-platform images and export cache
+      # https://github.com/docker/setup-buildx-action
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@f95db51fddba0c2d1ec667646a06c2ce06100226 # v3.0.0
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
```
Comment (CheesyLaZanya) [inline .github/workflows/docker-publish.yml:57]: Same for this, why not @v3?

File: .github/workflows/docker-publish.yml [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        id: meta
+        uses: docker/metadata-action@96383f45573cb7f253c731d3b3ab81c87ef81934 # v5.0.0
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      # Build and push Docker image with Buildx (don't push on PR)
+      # https://github.com/docker/build-push-action
+      - name: Build and push Docker image
+        id: build-and-push
+        uses: docker/build-push-action@0565240e2d4ab88bba5387d719585280857ece09 # v5.0.0
```
Comment (CheesyLaZanya) [inline .github/workflows/docker-publish.yml:75]: There looks to be a newer v6, any reason to be on v5?

https://github.com/marketplace/actions/build-and-push-docker-images#usage

**Review by CheesyLaZanya — APPROVED** [review]
Looks good to me!

**Pull request merged**